lib/autsr.gi

```diff
@@ -491,30 +539,148 @@ local S,c,hom,q,a,b,i,t,int,bad,have,ups,up,new,u,good,abort,clim,worked,pp,
 
   fi;
 
-  Info(InfoMorph,3,"Did ",tstcnt," tests");
+  Info(InfoMorph,3,"Did ",tstcnt," tests, grow by ",Index(S,S1));
   return S;
 
 end);
 
+# Same syntax as `SubgroupProperty`, but assumption that the tests are
+# expensive. Thus minimize number of tests by doing more group calculations
 InstallGlobalFunction(SubgroupConditionAbove,function(G,cond,Sorig)
-local cs,nr,u,no,un,S;
-  S:=Sorig;
-  cs:=CompositionSeries(G);
+local cs,nr,u,no,un,S,rad,res,ise,uno;
+
+
+  # first, try to find a few elements that work (and catch the case that the
+  # subgroup has small index. Ensure we test at least once for each possible
+  # step.
+  nr:=2^LogInt(Size(G),1000)+Length(Factors(Size(G)));
+  u:=[];
+  S:=SubgroupProperty(G,
+    function(elm)
+      if nr<0 then return true;fi;
+      nr:=nr-1;
+      if cond(elm) then
+        Add(u,elm);
+        return true;
+      else
+        return false;
+      fi;
+    end,Sorig);
+
+  if nr>=0 then # succeeded (small index case)
+    return S;
+  elif Length(u)>0 then
+    S:=ClosureGroup(Sorig,u);
+  else
+    S:=Sorig;
+  fi;
+
+  # now build along coposition series
```

Comment:
Obviously not important but you may as well fix if you're touching this file again...
```suggestion
  # now build along composition series
```